cumulus/parachains/runtimes/assets/asset-hub-rococo/tests/tests.rs

```diff
@@ -1471,3 +1471,28 @@ fn location_conversion_works() {
 		assert_eq!(got, expected, "{}", tc.description);
 	}
 }
+
+#[test]
+fn calling_payment_api_with_a_lower_version_works() {
+	use xcm::prelude::*;
+	use xcm_runtime_apis::fees::runtime_decl_for_xcm_payment_api::XcmPaymentApiV1;
+
+	ExtBuilder::<Runtime>::default().build().execute_with(|| {
+		let transfer_amount = 100u128;
+		let xcm_to_weigh = Xcm::<RuntimeCall>::builder_unsafe()
+			.withdraw_asset((Here, transfer_amount))
+			.buy_execution((Here, transfer_amount), Unlimited)
+			.deposit_asset(AllCounted(1), [1u8; 32])
+			.build();
+		let versioned_xcm_to_weigh = VersionedXcm::from(xcm_to_weigh.clone().into());
+		let lower_version_xcm_to_weigh =
+			versioned_xcm_to_weigh.into_version(XCM_VERSION - 1).unwrap();
+		let xcm_weight = Runtime::query_xcm_weight(lower_version_xcm_to_weigh);
+		assert!(xcm_weight.is_ok());
+		let native_token = VersionedAssetId::from(AssetId(Parent.into()));
+		let lower_version_native_token = native_token.into_version(XCM_VERSION - 1).unwrap();
+		let execution_fees =
+			Runtime::query_weight_to_asset_fee(xcm_weight.unwrap(), lower_version_native_token);
+		assert!(execution_fees.is_ok());
+	});
+}
```

Comment:
I think we don't need to copy this test, we could extract this as test_case here: https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachains/runtimes/assets/test-utils/src/test_cases.rs
and then reuse it here and later for fellows

Reply:
Also it would be very cool to catch and cover case where we hit `quote_price_tokens_for_exact_tokens` here